### DISCUSSION TITLE
Can a compiled function return an array where the length changes?

### DISCUSSION THREAD

**Discussion by cemlyn007** [ml-explore/mlx#757] [Q&A] [answered]
Trying to play around with the C++ API with a compiled function, I am getting cheeky segmentation fault!

Do compiled functions have to have static shaped arrays returned?

**Comment by cemlyn007** [comment]
I should add that I wouldn't be surprised if the compiled function inputs and outputs must have static shapes since libraries like JAX have this expectation but wondering because it would make my implementation easier! ;)

**Comment by awni** [comment, marked as answer]
> Trying to play around with the C++ API with a compiled function, I am getting cheeky segmentation fault!

Any interest in sharing the code ;). It shouldn't segfault if you do something wrong, just crash, so I would count that as a bug.

Shapes have to be the same (there is an experimental shapeless compile). If you change shapes then the function will be recompiled, so it's safe, but can be costly.

**Reply by cemlyn007** [reply]
Yeah I can try to share a minimal reproducible example tonight / tomorrow!

**Reply by cemlyn007** [reply]
I lost stashed code that had the segmentation fault so discard what I said!
